Cheat Sheet Poster

marcodallavecchia opened the pull request:
Closes #226 

Hello there! 
This is my first pull request ever and my first real way to contribute to the carpentries in general (also just finished the training sessions).

I was looking for ways to contribute to a data carpentry project as I'm into image analysis, so I've taken a look at this repo. Not finding much that needed immediate help, I decided to create a cheat sheet poster based on the Issue described [here](https://github.com/datacarpentry/image-processing/issues/226).

I have added a files folder in the root directory with the cheatsheet I created in HTML format. This can be directly opened and edited using [Draw.io](https://www.drawio.com/), I also included the exported PDF version and a couple of assets I created outside. I have used an asset from the [bioimagebook](https://bioimagebook.github.io/) by petebankhead citing its source, because I thought it would be cool, but I don't know if that is not allowed or might violate some policies. Let me know if that's the case.

I basically recreated the pictures of the whiteboard by @shaw2thefloor, and created an A4 horizontal page. I used Mulish font for the main text and titles, because that's the one I saw on the main course webpage, which can be added to draw.io like explained [here](https://www.drawio.com/blog/external-fonts). I also chose the green color trying to stick with the course website color palette.

Of course let me know if you think anything should be changed or if I've done something wrong! 
I hope this is useful and looking forward to more contributions!

tobyhodges commented:
Thanks so much for this contribution, @marcodallavecchia, and congratulations on opening your first pull request. I am grateful that you chose our project!

Overall I think this looks fantastic. I request only a few minor changes before I would be ready to merge it.

#### 1. Image licensing & credit

> I have used an asset from the [bioimagebook](https://bioimagebook.github.io/) by petebankhead citing its source, because I thought it would be cool, but I don't know if that is not allowed or might violate some policies.

That book (which looks like a very helpful resource) is [licensed CC-BY](https://bioimagebook.github.io/chapters/0-preamble/license.html), so we can re-use the images here (thank you @petebankhead!). I would like the acknowledgement of the source of that image to be more prominent, though. Please could you adjust the PDF to include a statement about where the image was taken from? E.g.

> Images marked with `*` are re-used or adapted from originals published in [_Introduction to Bioimage Analysis_](https://bioimagebook.github.io/index.html) by Pete Bankhead.

BTW I love that you have included the CC-BY license badge at the bottom-left of the cheatsheet 🔓❤️

#### 2. Accessibility
The colour contrast of the green text on white background looks quite low in the _RGB -- 3 channels_ section. Low contrast would present an accessibility challenge for folks with low vision and/or low colour vision. Please could you check the contrast and adjust the colour of the text to improve if needed? (Tools like [Color Contrast Checker](https://coolors.co/contrast-checker/112a46-acc8e5) can help with this.) It may be a good idea to check the contrast of the text on gray background too.

#### 3. Branding

> I also chose the green color trying to stick with the course website color palette.

Aside from the contrast improvements mentioned above, I am happy with the colours as they are. However, if you _want_ to make it match the Data Carpentry scheme, you can find details of the official palette at https://docs.carpentries.org/topic_folders/communications/resources/brand_identity.html#lesson-program-colours-and-shades 😆

shaw2thefloor commented:
@marcodallavecchia this is fantastic. Exactly what I had in mind. Thanks so much for putting this together.

shaw2thefloor commented:
Also, perhaps a link can be provided to this file in the preamble, and/or very near the beginning of the material, so people know it's there.

marcodallavecchia commented:
Hello again!
Firstly, thanks for the warm welcome and good feedback.

Secondly, I've updated the cheat sheet with the concerns from @tobyhodges.
1. I clearly stated the source at the bottom of the page, but used the # symbol rather than the * symbol as it was getting confusing with the convolution symbol already present
2. I improved the contrast of the RGB text and corresponding images matrices. I removed the gray box as white bg actually has a better contrast.
3. I update the banner color with the 'data carpentry' #205959 brand identity as suggested

The only question I would have is: where would it be best to add a link to this resource? Should it be on learner or instructor side? I couldn't find an obvious spot to add it, so I didn't yet. Feel free to suggest a place to add the line and I can do it.

Thanks a lot!

tobyhodges commented:
> Also, perhaps a link can be provided to this file in the preamble, and/or very near the beginning of the material, so people know it's there.

Good point, @shaw2thefloor. We should also mention it in the Instructor Notes so that Instructors know that it is available e.g. if they want to print it out before a workshop.

> The only question I would have is: where would it be best to add a link to this resource?

Good question, @marcodallavecchia. @shaw2thefloor's comment also set me to thinking about where the cheat sheet should be. I recommend that you move the files into the `episodes/files/` directory: this will make the page and PDF available within the lesson website. For example, if the cheatsheet is stored at `episodes/files/cheatsheet.html` it will be accessible at <https://datacarpentry.org/image-processing/files/cheatsheet.html>

Links can be added as mentions of the cheatsheet in:

* `episodes/02-image-basics.md`: I suggest a short `callout` after the opening paragraph
* `instructors/instructor-notes.md`: in a new section added underneath _Working with imageio and skimage_

marcodallavecchia commented:
I moved the cheat sheet in `episodes/files` and renamed the file to simply `cheatsheet`.

I added a link to the material into both the `instructor-notes.md` and `02-image-basics.md`. However, I'm not very familiar with how the website is compiled and didn't compile it locally to test it. Is the relative link I used going to work?

I used `[here](../episodes/files/cheatsheet.html)` markdown syntax. I guess that's better than using the direct URL i.e. `https://datacarpentry.org/image-processing/files/cheatsheet.html`, but let me know if that doesn't work as I think.

tobyhodges commented on instructors/instructor-notes.md:
```diff
@@ -36,6 +36,10 @@ to share your experience with the lesson Maintainers.
 
+## Additional resources
+
+- A cheat-sheet with graphics for some fundamentals of this course is available [here](../episodes/files/cheatsheet.html). A printable PDF version is [also available](../episodes/files/cheatsheet.pdf)
```
```suggestion
- A cheat-sheet with graphics illustrating some concepts in this lesson is available:
    - [Cheat-sheet HTML for viewing in browser](../episodes/files/cheatsheet.html). 
    - [PDF version for printing](../episodes/files/cheatsheet.pdf).
```

Adjusting wording to make the links more friendly to screen-reader users.

tobyhodges reviewed (changes requested):
This is looking excellent, @marcodallavecchia. I have suggested one slight adjustment to the structure of the Instructor Note. Once that is addressed, I will be delighted to approve and merge this.

tobyhodges reviewed (approved):
Thanks again @marcodallavecchia for this excellent contribution. I hope you'll consider making more in the future 🙌

tobyhodges commented:
Oh! I forgot @marcodallavecchia, please could you open another PR to add your details as a creator to [our zenodo.json file](https://github.com/datacarpentry/image-processing/blob/main/.zenodo.json)?

marcodallavecchia commented:
Sure, I will do that shortly! Thanks a lot!
